asttokens/asttokens.py

```diff
@@ -213,86 +245,160 @@ def get_tokens(self, node, include_extra=False):
     """
     return self.token_range(node.first_token, node.last_token, include_extra=include_extra)
 
-  def get_text_range(self, node):
-    # type: (AstNode) -> Tuple[int, int]
+  def get_text_positions(self, node, padded, unmarked=False):
+    # type: (AstNode, bool, bool) -> Tuple[Tuple[int, int], Tuple[int, int]]
     """
-    After mark_tokens() has been called, returns the (startpos, endpos) positions in source text
-    corresponding to the given node. Returns (0, 0) for nodes (like `Load`) that don't correspond
-    to any particular text.
+    Returns two ``(lineno, col_offset)`` tuples for the start and end of the given node.
+    If the positions can't be determined, or the nodes don't correspond to any particular text,
+    returns ``(1, 0)`` for both.
+
+    ``padded`` corresponds to the ``padded`` argument to ``astunparse.get_source_segment()``.
+    This means that if ``padded`` is True, the start position will be adjusted to include
+    leading whitespace if ``node`` is a multiline statement.
+
+    Use ``unmarked=True`` to force the method to not use token information.
+    This is not recommended for external use as it may raise errors or return incorrect results
+    for certain nodes. Prefer passing ``init_tokens=False`` to the constructor instead.
     """
+    if getattr(node, "_broken_positions", None):
+      return (1, 0), (1, 0)
+
+    node_dont_use_tokens = getattr(node, "_dont_use_tokens", None)
+    # Always use the unmarked method when unmarked=True.
+    if unmarked or (
+        # For nodes that work with the unmarked method, prefer it if either:
+        # 1. There are no tokens, i.e. `init_tokens=False` was passed to the constructor.
+        # 2. The node is better handled by the unmarked method. This applies to import aliases
+        #    and nodes within f-strings.
```

Comment:
Looks like this applies to more than just f-strings now?

Reply:
It seemed best to make `_dont_use_tokens` a generic name that could theoretically be applied to other things, but for now it only applies to nodes within f-strings.